staging/src/k8s.io/apiserver/pkg/admission/attributes.go

```diff
@@ -57,6 +58,13 @@ func (record *attributesRecord) GetNamespace() string {
 }
 
 func (record *attributesRecord) GetName() string {
+	// On create, get name from new object if unset in admission
+	if len(record.name) == 0 && record.GetOperation() == Create {
```

Comment:
I'd rather see this logic done where the record is constructed in the create rest handler... this object should remain really basic

Reply:
@liggitt Good suggestion.